unique_by_attr.py

```diff
@@ -0,0 +1,28 @@
+#! /usr/bin/env python3
+"""
+    If we have some sequence of objects and want to remove items with the same attribute value
+    Python creates a dict, where keys are value if attribute (bar in our case), values are object of the sequence.
+    After that the dict is transformed back to list
+
+    Note: in result we save the last from repeating elements (item2 in our case)!
+"""
+
+
+class Foo():
+    pass
```

Comment:
Could you change "bar" to be a class attribute and set it's value with the constructor?